src/common/libflux/handle.c

```diff
@@ -1140,16 +1140,16 @@ int flux_pollevents (flux_t *h)
 {
     h = lookup_clone_ancestor (h);
     int e, events = 0;
+    struct epoll_event ev;
 
     /* create pollfd if needed */
     if (flux_pollfd (h) < 0)
         return -1;
 
-    /* wait for handle event */
-    if (h->pollfd >= 0) {
-        struct epoll_event ev;
-        (void)epoll_wait (h->pollfd, &ev, 1, 0);
-    }
+    /* consume all epoll events before sampling pollevents */
+    while (epoll_wait (h->pollfd, &ev, 1, 0) == 1)
+        ;
+
```

Comment:
Is there any possibility this loop could run forever if there is a steady stream of messages or other activity on this handle from the other side? I do not know this section of the code that well and did not dig into it, so apologies if this question is naive.